lib/ansible/modules/network/haproxy.py

```diff
@@ -287,8 +307,11 @@ def get_state_for(self, pxname, svname):
         """
         data = self.execute('show stat', 200, False).lstrip('# ')
         r = csv.DictReader(data.splitlines())
-        state = tuple(map(lambda d: { 'status': d['status'], 'weight': d['weight'] }, filter(lambda d: (pxname is None or d['pxname'] == pxname) and d['svname'] == svname, r)))
-        return state or None
+        state = []
+        for d in r:
+            if (pxname is None or d['pxname'] == pxname) and d['svname'] == svname:
+                state.append({ 'status': d['status'], 'weight': d['weight'], 'scur': d['scur'] })
+        return state
```

Comment:
Not sure if it matters but prior code had `return state or None` which will now not be `None`; might not matter, but it's a functional difference.